pcdsdevices/pim.py

```diff
@@ -45,14 +47,22 @@ class PIMY(InOutRecordPositioner, BaseInterface):
     _icon = 'fa.camera-retro'
 
     tab_whitelist = ['stage']
+    _pre_stage_state = None
 
-    def stage(self):
+    def stage(self) -> list[OphydObject]:
         """Save the original position to be restored on `unstage`."""
-        self._original_vals[self.state] = self.state.get()
+        self._pre_stage_state = self.state.get()
+        return super().stage()
+
+    def unstage(self) -> list[OphydObject]:
+        """Load the original position that was saved on `stage`."""
+        if self._pre_stage_state is not None:
+            self.state.put(self._pre_stage_state)
+        self._pre_stage_state = None
         return super().stage()
```

Comment:
```suggestion
        return super().unstage()
```

Makes me think we might need a "generically try to stage/unstage all fake device classes" test

Reply:
Oops, this one even passes the PIM stage/unstage test though!